transformer_engine/common/__init__.py

```diff
@@ -8,35 +8,53 @@
 import functools
 import glob
 import importlib
-from importlib.metadata import version, metadata, PackageNotFoundError
-import logging
+from importlib.metadata import version, distribution, PackageNotFoundError
 import os
 from pathlib import Path
 import platform
 import subprocess
 import sys
 import sysconfig
-from typing import Optional
-
-
-_logger = logging.getLogger(__name__)
+from typing import Optional, Tuple
 
 
 @functools.lru_cache(maxsize=None)
-def _is_pip_package_installed(package) -> bool:
+def _is_package_installed(package) -> bool:
     """Check if the given package is installed via pip."""
 
     # This is needed because we only want to return true
     # if the python package is installed via pip, and not
     # if it's importable in the current directory due to
     # the presence of the shared library module.
     try:
-        metadata(package)
+        distribution(package)
     except PackageNotFoundError:
         return False
     return True
 
 
+@functools.lru_cache(maxsize=None)
+def _is_package_installed_via_pypi(package) -> bool:
+    """Check if the given package is installed via PyPI."""
+
+    if not _is_package_installed(package):
+        return False
+
+    te_dist = distribution("transformer-engine")
```

Comment:
Do we need to check that this `transformer-engine` package matches the currently loaded `transformer_engine` module? It's possible that the user has multiple TE installs (e.g. the system comes with a PyPI system install in `/usr/local/lib/python3.12/dist-packages` and a non-root user has built TE locally in `~/.local/lib/python3.12/site-packages`).